Save and restore output properties "Broadcast RGB" and "audio"

lkrms opened the pull request:
`Broadcast RGB` defaults to `Automatic`, which in my experience can mean `Limited 16:235` even when hardware is capable of `Full`, leading to washed out or excessively bright output. autorandr's lack of support for output properties managed via `xrandr --set` forced me to find a much less elegant solution, but now I'm proposing this fix instead.

(Persisting the `audio` property is far less important, but disabling display audio automatically will make my life simpler, so I've included that as well.)

These options represent multiple possible `--set` arguments, each with a property name and value, either of which may contain spaces. Lines in `~/.config/autorandr/*/config` only map one xrandr argument to one value, so the new values are being stored as `broadcast-rgb` and `audio` respectively. To avoid a massive re-work of code that touches `*.options` in `autorandr.py`, these "options" aren't converted to `--set <property> <value>` until the last possible moment, i.e. in `option_vector`.

This means `--skip-options` values will need to correspond to stored names (`broadcast-rgb`) rather than the final argument name (i.e. `set`, which would be ambiguous anyway).

phillipberndt commented:
Sorry it took me so long, this PR somehow slipped through..

Since properties are somewhat dynamic, how about prefixing `prop--` and managing them via a list rather than manually? I'm thinking

    properties = ["aspect ratio", "Broadcast RGB", "audio", "non-desktop",]

and then in the regex insert a `properties.join("|")`, and everywhere else loop through this array, and store as `prop-audio` etc.

lkrms commented:
@phillipberndt I'm not sure if you're suggesting `properties` be user-configurable, but assuming that's not the case, I fully agree that a more dynamic approach would be better, and I've just pushed a commit that implements your suggestion.  

Unfortunately it wasn't quite as simple as "|".join(properties), thanks to `re` only returning the last of any repeated subpattern matches, but hopefully it's a step closer to a result that works for everybody.

I've included a couple of additional properties after checking they're widely used with `--set` (largely by browsing forums).

lkrms commented:
@phillipberndt I've just rebased and squashed this branch to simplify things. Please let me know if there's something you'd like handled differently :smiley:

phillipberndt commented on autorandr.py:
```diff
@@ -176,6 +185,7 @@ class XrandrOutput(object):
             CRTC:\s*(?P<crtc>[0-9]) |                                                   # CRTC value
             Transform: (?P<transform>(?:[\-0-9\. ]+\s+){3}) |                           # Transformation matrix
             EDID: (?P<edid>\s*?(?:\\n\\t\\t[0-9a-f]+)+) |                               # EDID of the output
+            %s |                                                                        # Properties to include in the profile
```
I think in this particular case I'd prefer `""" + foo + """`. Yes, that's slow in Python, but this is a giant string, and imho readability of the overall thing counts here.

phillipberndt commented on autorandr.py:
```diff
@@ -185,7 +195,9 @@ class XrandrOutput(object):
         )*)
-    """
+    """ % ("|".join([r"{}:\s*(?P<{}>[\S ]*\S+)"
+                     .format(re.sub(r"(\s)", r"\\\1", p),
```
What is the significance of `re.sub(r"(\s)", r"\\\1", p)`?

phillipberndt commented on autorandr.py:
```diff
@@ -387,6 +411,9 @@ def from_xrandr_output(cls, xrandr_output):
                 options["rate"] = match["rate"]
+            for prop in [re.sub(r"\W+", "_", p.lower()) for p in properties]:
+                if match[prop]:
+                    options["prop-" + prop] = match[prop]
```
One thing is bad about this approach: It breaks with the tradition of having a config file where translation into xrandr arguments is very simple. I wonder if we should do something to keep that, or at least flag this as something special? (Like prefixing another `x-`..)

That's a genuine question, please don't treat this as a change request :)

phillipberndt reviewed (commented):
Gosh, I'm so sorry for missing yet another revision 🤨 Thank you for being persistent about this and reminding me!!

I have one major comment/question. See details.

phillipberndt commented:
Since you didn't respond to the `x-` question but kept pushing never revisions I went ahead and merged this with a patch on top to prepend `x-`.

Thanks again for your contribution!

lkrms commented:
Hi @phillipberndt, thank you! I've been snowed under by work/life/health/COVID/lockdown/everything, so although I actioned most of what you suggested and did a rebase/push, following up on the PR languished on my to do list (like way too many other things). Look, I have proof!

![Screen Shot 2021-12-16 at 21 35 25](https://user-images.githubusercontent.com/2563783/146355692-f4bbe178-7d0d-43f0-84e7-c77ea9784cf8.png)

Thanks for merging this and being kind with my silence/withdrawal. I hope you're well 😃
